Buildings/Applications/DHC/EnergyTransferStations/CoolingDirectControlledReturn.mo

```diff
@@ -0,0 +1,396 @@
+       controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  parameter Real Ni(
+    final min=100*Modelica.Constants.eps)=0.9
+    "Ni*Ti is time constant of anti-windup compensation"
+    annotation(Dialog(tab="Controller", enable=
+       controllerType == Modelica.Blocks.Types.SimpleController.PI or
+       controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  parameter Real Nd(
+    final min=100*Modelica.Constants.eps)=10
+    "The higher Nd, the more ideal the derivative block"
+    annotation(Dialog(tab="Controller", enable=
+       controllerType == Modelica.Blocks.Types.SimpleController.PD or
+       controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  parameter Modelica.Blocks.Types.InitPID initType=
+    Modelica.Blocks.Types.InitPID.DoNotUse_InitialIntegratorState
+    "Type of initialization (1: no init, 2: steady state, 3: initial state, 
+      4: initial output)"
+    annotation(Dialog(group="Initialization",tab="Controller"));
+
+  parameter Real xi_start=0
+    "Initial or guess value value for integrator output (= integrator state)"
+    annotation (Dialog(group="Initialization",
+      tab="Controller", enable=
+        controllerType == Modelica.Blocks.Types.SimpleController.PI or
+        controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  parameter Real xd_start=0
+    "Initial or guess value for state of derivative block"
+    annotation (Dialog(group="Initialization",
+      tab="Controller", enable=
+        controllerType == Modelica.Blocks.Types.SimpleController.PD or
+        controllerType == Modelica.Blocks.Types.SimpleController.PID));
+
+  parameter Real yCon_start=0
+    "Initial value of output from the controller"
+    annotation(Dialog(group="Initialization",
+      tab="Controller",
+      enable=initType == Modelica.Blocks.Types.InitPID.InitialOutput));
+
+  parameter Boolean reverseAction = true
+    "Set to true for throttling the water flow rate through a cooling 
+      coil controller"
+    annotation(Dialog(tab="Controller"));
+
+  // Advanced parameters
+  parameter Modelica.Fluid.Types.Dynamics energyDynamics=
+    Modelica.Fluid.Types.Dynamics.FixedInitial
+    "Type of energy balance: dynamic (3 initialization options) or steady state"
+    annotation(Dialog(tab="Advanced"));
+
+  parameter Modelica.SIunits.PressureDifference[3] dp_nominal=500*{1,-1,1}
+    "Nominal pressure drop in pipe junctions"
+    annotation(Dialog(tab="Advanced"));
+
+  Modelica.Blocks.Interfaces.RealInput TSetDisRet
+    "Setpoint for the minimum district return temperature"
+    annotation (Placement(transformation(extent={{-140,-140},{-100,-100}})));
+
+  Modelica.Blocks.Interfaces.RealOutput Q_flow(
+    final quantity="Power",
+    final unit="W",
+    displayUnit="kW")
+    "Measured power demand at the ETS"
+    annotation (Placement(transformation(extent={{100,140},{120,160}})));
+
+  Modelica.Blocks.Interfaces.RealOutput Q(
```

Comment:
@AntoineGautier I like both of your suggestions here. A block would be nice since we are seeing similar post processing quantities across ETS, plant, and network connection models. Have you made any block(s) of this sort for the models you've been working on? I'll hold off on making these changes until we've agreed upon a plan.

Reply:
I added a boolean flag `show_heaFlo` to be consistent with `Buildings.Applications.DHC.Networks.BaseClasses.PartialConnection1Pipe.mo`. I have not yet made all postprocessing into a block and will still plan to discuss with you soon.